x-pack/legacy/plugins/maps/public/layers/fields/es_doc_field.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FIELD_ORIGIN } from '../../../common/constants';
+import { ESTooltipProperty } from '../tooltips/es_tooltip_property';
+import { ITooltipProperty, TooltipProperty } from '../tooltips/tooltip_property';
+import { COLOR_PALETTE_MAX_SIZE } from '../../../common/constants';
+import { indexPatterns } from '../../../../../../../src/plugins/data/public';
+import { IFieldType } from '../../../../../../../src/plugins/data/public';
+import { IField, AbstractField } from './field';
+import { IESSource } from '../sources/es_source';
+import { IVectorSource } from '../sources/vector_source';
+
+export class ESDocField extends AbstractField implements IField {
+  static type = 'ES_DOC';
+
+  private readonly _source: IESSource;
+
+  constructor({
+    fieldName,
+    source,
+    origin,
+  }: {
+    fieldName: string;
+    source: IESSource;
+    origin: FIELD_ORIGIN;
+  }) {
+    super({ fieldName, origin });
+    this._source = source;
+  }
+
+  canValueBeFormatted(): boolean {
+    return true;
+  }
+
+  getSource(): IVectorSource {
+    return this._source;
+  }
+
+  async _getIndexPatternField(): Promise<IFieldType | undefined> {
+    const indexPattern = await this._source.getIndexPattern();
+    const indexPatternField = indexPattern.fields.getByName(this.getName());
+    return indexPatternField && indexPatterns.isNestedField(indexPatternField)
+      ? undefined
+      : indexPatternField;
+  }
+
+  async createTooltipProperty(value: string | undefined): Promise<ITooltipProperty> {
+    const indexPattern = await this._source.getIndexPattern();
+    const tooltipProperty = new TooltipProperty(this.getName(), await this.getLabel(), value);
+    return new ESTooltipProperty(tooltipProperty, indexPattern, this as IField);
+  }
+
+  async getDataType(): Promise<string> {
+    const indexPatternField = await this._getIndexPatternField();
+    return indexPatternField ? indexPatternField.type : '';
+  }
+
+  supportsFieldMeta(): boolean {
+    return true;
+  }
+
+  async getOrdinalFieldMetaRequest(): Promise<unknown> {
+    const indexPatternField = await this._getIndexPatternField();
+
+    if (
+      !indexPatternField ||
+      (indexPatternField.type !== 'number' && indexPatternField.type !== 'date')
+    ) {
+      return null;
+    }
+
+    const extendedStats: any = {};
```

Comment:
I'd avoid using `any` especially since we're specifically assigning just a few lines later.

I'd do something like:     `const extendedStats: { script?: unknown; field?: unknown } = {};`